x/ibc-rate-limit/ibc_middleware_test.go

```diff
@@ -0,0 +1,763 @@
+func (suite *MiddlewareTestSuite) TestReceiveTransferNoContract() {
+	suite.ConfigureTransferChannel()
+	one := sdkmath.NewInt(1)
+	_, err := suite.AssertReceive(true, suite.MessageFromBToA(sdk.DefaultBondDenom, one))
+	suite.Require().NoError(err)
+}
+
+func (suite *MiddlewareTestSuite) initializeEscrow() {
+	app := suite.GetNeutronZoneApp(suite.ChainA)
+	supply := app.BankKeeper.GetSupply(suite.ChainA.GetContext(), sdk.DefaultBondDenom)
+
+	// Move some funds from chainA to chainB so that there is something in escrow
+	// Each user has 10% of the supply, so we send most of the funds from one user to chainA
+	transferAmount := supply.Amount.QuoRaw(20)
+
+	// When sending, the amount we're sending goes into escrow before we enter the middleware and thus
+	// it's used as part of the channel value in the rate limiting contract
+	// To account for that, we subtract the amount we'll send first (2.5% of transferAmount) here
+	sendAmount := transferAmount.QuoRaw(40)
+
+	// Send from A to B
+	_, _, err := suite.FullSendAToB(suite.MessageFromAToB(sdk.DefaultBondDenom, transferAmount.Sub(sendAmount)))
+	suite.Require().NoError(err)
+	// Send from B to A
+	_, _, err = suite.FullSendBToA(suite.MessageFromBToA(sdk.DefaultBondDenom, transferAmount.Sub(sendAmount)))
+	suite.Require().NoError(err)
+}
+
+func (suite *MiddlewareTestSuite) fullSendTest(native bool) map[string]string {
+	quotaPercentage := 5
+	suite.initializeEscrow()
+	// Get the denom and amount to send
+	denom := sdk.DefaultBondDenom
+	if !native {
+		denomTrace := transfertypes.ParseDenomTrace(transfertypes.GetPrefixedDenom("transfer", channelTest, denom))
+		fmt.Println(denomTrace)
+		denom = denomTrace.IBCDenom()
+	}
+
+	app := suite.GetNeutronZoneApp(suite.ChainA)
+
+	// This is the first one. Inside the tests. It works as expected.
+	channelValue := CalculateChannelValue(suite.ChainA.GetContext(), denom, app.BankKeeper)
+
+	// The amount to be sent is send 2.5% (quota is 5%)
+	quota := channelValue.QuoRaw(int64(100 / quotaPercentage))
+	sendAmount := quota.QuoRaw(2)
+
+	fmt.Printf("Testing send rate limiting for denom=%s, channelValue=%s, quota=%s, sendAmount=%s\n", denom, channelValue, quota, sendAmount)
+
+	// Setup contract
+	testOwner := sdk.MustAccAddressFromBech32(testutil.TestOwnerAddress)
+	suite.StoreTestCode(suite.ChainA.GetContext(), testOwner, "./bytecode/rate_limiter.wasm")
+	quotas := suite.BuildChannelQuota("weekly", channelTest, denom, 604800, 5, 5)
+	addr := suite.InstantiateRLContract(quotas)
+	suite.RegisterRateLimitingContract(addr)
+
+	// send 2.5% (quota is 5%)
+	fmt.Printf("Sending %s from A to B. Represented in chain A as wrapped? %v\n", denom, !native)
+	_, err := suite.AssertSend(true, suite.MessageFromAToB(denom, sendAmount))
+	suite.Require().NoError(err)
+
+	// send 2.5% (quota is 5%)
+	fmt.Println("trying to send ", sendAmount)
+	r, _ := suite.AssertSend(true, suite.MessageFromAToB(denom, sendAmount))
```

Comment:
errors are not handles, fix it everywhere, please

Reply:
there is no need for that as soon as `AssertSend` designed to handle (or expect) the error depending on the first parameter